.buildkite/pipeline.yml

```diff
@@ -10,37 +10,31 @@ steps:
     if: &not_scheduled_autodoc
       build.branch != "autodoc/master" && build.branch != "master" && build.source != "schedule"
     commands:
-    - nix run -f ci.nix pkgs.hlint -c
+    - nix shell .#haskellPackages.hlint -c
         ./scripts/lint.sh
 
-  - label: reuse lint
+  - label: check
     if: *not_scheduled_autodoc
     commands:
-    - nix run -f ci.nix pkgs.reuse -c
-        reuse lint
-
-  - label: check trailing whitespace
-    if: *not_scheduled_autodoc
-    commands:
-    - .buildkite/check-trailing-whitespace.sh
+    - nix flake check -L
 
   - label: xrefcheck
     if: *not_scheduled_autodoc
     commands:
-    - nix shell -f ci.nix pkgs.xrefcheck -c xrefcheck
+    - nix shell .#xrefcheck --impure -c xrefcheck
```

Comment:
I suggest adding a TODO to remove `--impure` once `xrefcheck` is flakified, otherwise there's a chance this `--impure` will stay forever.